src/aposteriori/data_prep/create_frame_data_set.py

```diff
@@ -1266,13 +1329,22 @@ def make_frame_dataset(
         into a 3x3x3 gaussian density using the formula indicated by Zhang et al., (2019) ProdCoNN.
 
         https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1002%2Fprot.25868&file=prot25868-sup-0001-AppendixS1.pdf
-
+    blacklist_csv: StrOrPath
+        Path to blacklist csv file.
 
     Returns
     -------
     output_file_path: pathlib.Path
         A path to the location of the output dataset.
     """
+    # Filter by blacklist:
+    if blacklist_csv and pathlib.Path(blacklist_csv).exists():
+        structure_files = filter_structures_by_blacklist(
+            structure_files, pathlib.Path(blacklist_csv)
+        )
+    elif blacklist_csv and pathlib.Path(blacklist_csv).exists() == False:
+        raise InputError(f"Blacklist Path {blacklist_csv} not found.")
```

Comment:
Again, this is prone to bugs, especially as the rare execution path is not tested. I'd simplify this a bit by changing the `structure_files` function argument to `input_structure_files` then changing this to:

```suggestion
    if blacklist_csv:
        if pathlib.Path(blacklist_csv).exists():
            structure_files = filter_structures_by_blacklist(
                input_structure_files, pathlib.Path(blacklist_csv)
            )
        else:
            raise InputError(f"Blacklist Path {blacklist_csv} not found.")
    else:
        structure_files = input_structure_files
```
It's generally a good idea to avoid nested if statements, but in this case I think it makes it clearer and much more difficult to introduce a bug in the logic.